x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java

```diff
@@ -688,6 +690,14 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
         Files.deleteIfExists(keystoreBackupPath);
     }
 
+    protected String httpHostSettingValue(InetAddress[] allAddresses) {
```

Comment:
There are no tests added but: 

- I don't think we can control the interfaces used in our packaging tests so that we can end up with only local addresses
- A unit test for `AutoConfigureNode#httpHostSettingValue` would basically be a unit test for `InetAddress#isSiteLocalAddress`

Reply:
I think this is the best we can do, as you say.
The `NetworkService` doesn't have a public static method to resolve bind addresses of different types, for good reasons, I'm sure.

When we write to the node configuration we rely on the node to interpret the settings in a certain way, and we can't always use the node code that does that, neither we can always test the interaction.